### DISCUSSION TITLE
Alarm Behavior: Should it pause other audio players?

### DISCUSSION THREAD

**Discussion by gdelataillade** [gdelataillade/alarm#104] [General]
When the alarm is activated, how should it interact with other active audio players?

1) Pause the other audio players and then resume them after the alarm stops?
2) Play simultaneously with other audio sources?
3) Diminish the volume of other audio sources during the alarm ring?

Please share your thoughts and suggestions on the best approach.

**Comment by synstin** [comment]
@gdelataillade

I want it to ring at the same time as other audio sources.
Or at least have a field to set it.

**Reply by gdelataillade** [reply]
Hi @synstin,

Thanks for your feedback. I'm addressing your concern with the assumption that it's about the Android version of the plugin.

I'm currently working hard on version `3.0.0`. In this update, if `loopAudio` is set to false, your other audio source (music, YouTube video...) will have its volume reduced while the ringtone plays. After the ringtone ends, the volume will return to normal. For instance, if your ringtone is a 3-second audio file, the other audio source's volume will be lowered for those 3 seconds (but it will continue playing), and then it will return to its original level.

I hope this is what you're looking for. You can test this feature right now in the development version, `3.0.0-dev.8`, available here:
https://pub.dev/packages/alarm/versions/3.0.0-dev.8.
Stay tuned, as I'm hoping to officially release `3.0.0` very soon!

Please let me know if there's anything else you need, and I'll continue to do my best to provide the features your users require.

**Reply by pyjserv** [reply]
Hi @gdelataillade 

Thanks for your work. This sounds like setting a lower volume for other players when the alarm event starts. If `loopAudio` is set to true, do the other players stay at a reduced volume until the alarm is stopped?

**Reply by synstin** [reply]
@gdelataillade 

Thanks for the answer!
Is it possible to make it so that the volume of the other audio is not temporarily reduced, but has no effect on the other audio?

**Reply by synstin** [reply]
@gdelataillade 

Or you said it goes down for 3 seconds, and I'd like to be able to control that number of seconds for both IOS and Android.

**Reply by gdelataillade** [reply]
Hi @pyjserv @synstin 

If `loopAudio` is set to true, the audio will play forever until alarm is stopped. Other audios will have their volume reduce until alarm is stopped.

If `loopAudio` is set to false, the audio will play once and not repeat. The duration of the ringtone is the duration of the audio mp3 file provided as `assetAudioPath`. During this duration, other audios will have their volume reduced but will keep playing. When this duration is passed, other audios will have their volume return to their original levels.

Let me know if it's not clear or if you have any other questions.

**Reply by pyjserv** [reply]
Hi, @gdelataillade 

I tested 3.0.0-dev.8 and found that when the alarm event starts, the sound from other players is still quite noticeable. Is it possible to make the sound from other players even quieter? Also, I noticed that the `stopOnNotificationOpen` setting has been removed. Could you bring back this setting?

**Reply by gdelataillade** [reply]
Hi @pyjserv 

No, I'm not able to make the sound from other players quieter. It's managed by the operating system. [Source](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK)

I removed `stopOnNotificationOpen` because I believe it's something you could do manually, on the app side, with listeners. Is it OK for you ?

**Reply by pyjserv** [reply]
Hi @gdelataillade 

> No, I'm not able to make the sound from other players quieter. It's managed by the operating system. [Source](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK)

Understood.🙂

> I removed stopOnNotificationOpen because I believe it's something you could do manually, on the app side, with listeners. Is it OK for you ?

One approach I can think of is to check if the alarm has triggered every time the app resumes. Is there a better way?

Either way, `3.0.0` looks fantastic! Thank you so much for your work!

**Reply by gdelataillade** [reply]
> One approach I can think of is to check if the alarm has triggered every time the app resumes. Is there a better way?

I was thinking the same way !

**Comment by pyjserv** [comment]
>3. Diminish the volume of other audio sources during the alarm ring?

Hi @gdelataillade 

Thank you for your work. I personally prefer 3, which is also the default behavior for the system alarm. However, either 1 or 3 is okay, regardless of whether `loopAudio` is set to false or true.

**Comment by JosipDomazetDev** [comment]
Thank you for your plugin.
In my opinion either 1 or 3 is the way to go. Although 3 seems to be a little "hackish" especially if it actually adjusts the user volume. 

For my use case the most essential feature is that the normal audio (e.g. Spotify) resumes after the alarm.
